src/main/scala/io/iohk/ethereum/blockchain/sync/RegularSync.scala

```diff
@@ -361,11 +419,20 @@ class RegularSync(
         log.debug(s"got new blocks up till block: ${importedBlocks.last.header.number} " +
           s"with hash ${Hex.toHexString(importedBlocks.last.header.hash.toArray[Byte])}")
       }
+
       errorOpt match {
+        case Some(missingNodeEx: MissingNodeException) =>
+          log.error(missingNodeEx, "Requesting missing state nodes")
+          headersQueue = headersQueue.drop(blocks.length)
+          missingStateNodeRetry = Some(MissingStateNodeRetry(missingNodeEx.hash, peer, m))
```

Comment:
shouldn't we set `MissingStateNodeRetry#bodies` the blocks that weren't executed (because it stopped when missing MPT error occurred) instead of `m`?

Reply:
yes, also `headersQueue` was wrong, and that caused the unnecessary re-download of block headers and bodies to be retried